iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/services/Devices.java

```diff
@@ -510,4 +510,8 @@ public DeviceTwinName getDeviceTwinNames()
             }
         );
     }
+
+    public void finalize(){
```

Comment:
For important bug fixes like this one I generally prefer to have a separate PR for them. Its okay in this case but it helps for visibility / tracking.

Reply:
I think this is a temporary one and we need major refactoring of the code for that function.

Reply:
Yeah this is okay because its at least in a separate commit.